v1 Integration

jonathansick opened the pull request:
**Published at: https://ldm-493.lsst.io/v/v1**

This PR is for comments by the DMLT (and TCT) in preparation for acceptance by the DM TCT. Once accepted, this PR will be merged to `master`.

jonathansick committed:
Change notice: branch is not under change control

jonathansick committed:
Add summary section

jonathansick committed:
Documentation taxonomy

- outline types of documents
- diagram an idealized information flow across documents

jonathansick committed:
WIP: publishing technologies

jonathansick committed:
WIP: formats

jonathansick committed:
Describe landing pages

jonathansick committed:
Describe LTD

jonathansick committed:
WIP: describe DocHub

jonathansick committed:
Re-organize document into partial files

This will make the document easier to maintain.

jonathansick committed:
Add text on subjects out of scope

jonathansick committed:
Add section on licensing policy

jonathansick committed:
Introduce techdocs team and distribution of work

jonathansick committed:
Discussion about citeability (DOI, ADS, DocuShare)

jonathansick committed:
Add detailed discussion of user guides

jonathansick committed:
Add description of technote doc class

jonathansick committed:
Add implementation technote section

jonathansick committed:
Use full link to LPM-162

The link check was failing on the ls.st forward.

jonathansick committed:
Add discussion around LDM change controlled docs

jonathansick committed:
Describe single-page Sphinx sites

jonathansick committed:
Add discussion about Sphinx user guides

jonathansick committed:
Add Sphinx introduction

jonathansick committed:
Clean up and links

jonathansick committed:
Markup glitches

frossie committed:
Minor edits for clarity

frossie committed:
For substantive comments see full commit message

Responsibility lies with a person, and we already have identified (in
reality) who that person is - the Documentation Engineer (@jsick at this
time). I don't object to strongly to abstracting it to a TechDocs team
terminology if that makes it map industry conecpts (eg. the Twitter
reference) if that brings useful semantics to the conversation, but
ultimately any effort expended here is simply a scaling of the
Documentation Engineer.

frossie committed:
Relatively minor edits for clarity

frossie committed:
Relatively minor edits for clarity

frossie committed:
Substantive comment in the long commit message.

In general I am very uncomfortable with suggesting that we accept Google
Docs and Wiki pages as primary documentation formats. I can think of a
number of reasons where this is a sub optimal. I think a case should
need to be made for the workflows that are described here to apply. Or
really, we could be Nazis and just say DocHub may provide discovery but
we do not curate the content in LSST-The-Docs (frankly I am leaning that
way and would like to hear arguments as to the contrary).

frossie committed:
Remove implementation notes

frossie committed:
Remove implementation notes (fix remaining ref)

frossie committed:
More implementation link removal

jonathansick committed:
Remove Implementation Technotes from flow diagram

jonathansick committed:
Change-controlled documentation workflow

- Defer to TCT documentation in LDM-294
- Clarify classified/un-classified workflows
- Only Sphinx and LaTeX documents are hosted on LTD; all others are only
  served from DocuShare.

jonathansick committed:
Edits

- remove mentions to TechDocs team
- other small fixes

jonathansick committed:
Wrap linkcheck in travis_retry

This should insulate the link check from transient errors.

jonathansick committed:
Copy edits

jonathansick committed:
Make DocuShare a publishing platform

- Word/Google/etc docs are not longer shimmed through the Landing Page
  Framework. Landing pages are only for documents managed natively
  through GitHub. Office documents are delivered to docushare.
- Confluence pages must be converted to a format published by LTD or
  DocuShare before being considered delivered.
- Office documents must be delivered in DocuShare in order to be
  considered delivered.

jonathansick committed:
Copy edits

jonathansick committed:
Merge branch 'tickets/DM-7572' into v1

ktlim commented on _ldm.rst:
```diff
@@ -0,0 +1,40 @@
+
+Change-controlled design documentation defines the scope and budget of the Data Management System.
+DM's change-controlled design documents carry "LDM" handles.
+New and revised LDM documents are reviewed by the DM Technical Control Team (TCT), as described in `LDM-294: Data Management Organization and Charter`_, and submitted to the LSST Project `Change Control Board <LPM-19: Change Control Process>`_ for approval.
```
LDM doesn't go to the CCB, only the TCT and the DM Lead.  LSE and LPM documents are controlled by the CCB; those are what defines the scope (and indirectly budget) of the Data Management System.

ktlim commented on _ldm.rst:
```diff
@@ -0,0 +1,40 @@
+
+Some change-controlled documents are only available through :ref:`DocuShare <docushare>`, such as :ref:`word-processing files and spreadsheets <office>`, and all *classified* (non-public) documents.
+In these cases, :ref:`DocHub <platforms-dochub>` indexes and links to the document from :ref:`DocuShare <docushare>`.
+These documents are not be mirrored on :ref:`LSST the Docs <platforms-ltd>`.
```
Typo.

jonathansick replied:
Thanks; fixed in `tickets/DM-8099`.

ktlim commented on _ldm.rst:
```diff
@@ -0,0 +1,40 @@
+In these cases, :ref:`DocHub <platforms-dochub>` indexes and links to the document from :ref:`DocuShare <docushare>`.
+These documents are not be mirrored on :ref:`LSST the Docs <platforms-ltd>`.
+
+DM members are encouraged to author documents in formats that can be version-controlled in Git and published by :ref:`LSST the Docs <platforms-ltd>` for improved collaboration.
```
It's somewhat arguable as to whether git-based workflows provide improved collaboration over Confluence or Google Docs commenting and multi-user editing.

jonathansick replied:
That's a fair opinion. I'll re-phrase as saying that Git-based workflows are better integrated with JIRA and the DM documentation platforms.

ktlim commented on _taxonomy.rst:
```diff
@@ -0,0 +1,69 @@
+User Guides (DMG)
+   These documentation products describe usage of DM software, platforms, and data products to end-users.
+   Typically, end-users are astronomers in the scientific community.
+   Some **user guides** may instead be considered internal, such as **operational guides** or **developer guides**.
```
It's not immediately obvious to me that operator manuals for services fit into the same bucket as user guides.

jonathansick replied:
In what way do you see operational manuals as distinct from user guides?

The basic axes in the documentation taxonomy are:
1. **Authoring policy/controls.** User guides are written by the DM team as part of work tickets. I imagine the same is true for operational manuals.
2. **Format.** User guides are multi-page sphinx websites. I imagine that operational manuals will be presented the same way.
3. **Access.** User guides are written and published in the open. I suppose there could be some need to require authentication to see operations guides. If so we should identify that so we can plan to build a 'secured' variant of LSST the Docs.

Does this capture your concern? Or are you thinking of different differences?

ktlim commented on _taxonomy.rst:
```diff
@@ -0,0 +1,69 @@
+   See :ref:`ldm` for how change-controlled documents are handled by the Data Management Documentation Architecture.
+
+Technical notes (DMTN, SQR)
+   Technical notes are a standardized, durable, document series allowing useful information to be published on demand by any Data Management team member wishing to capture and share knowledge.
```
We may need another group for things like test plans and verification documentation.  I'm not sure that DMTNs would be adequate for this.

It's also not clear where lower-level, non-change-controlled design documentation fits.  Some of this should be with the code, but not necessarily in user guides (although maybe this is really what "developer guides" means).

jonathansick replied:
> We may need another group for things like test plans and verification documentation. I'm not sure that DMTNs would be adequate for this.

Can **test plans** be written as LDM documents?

Code that carries out test measurements will be documented with User Guides, and the LDM document and user guides can link between each other.

**Verification documentation** is an interesting subject. In [Figure 1](https://ldm-493.lsst.io/v/v1/index.html#fig-information-flow) I envisioned a class of 'validation documentation' but I didn't include this in the taxonomy or in a dedicated section because I'm not sure what the requirements for this documentation will be. @frossie has talked about dynamically-generating validation/verification documentation as runnable Jupyter notebooks. We can certainly handle this with the existing infrastructure. Systems engineering might have additional requirements and we should engage with them.

---

Regarding lower-level, non-change-controlled design documentation: does https://ldm-493.lsst.io/v/v1/index.html#relation-between-users-guides-and-design-documentation answer your question? tl;dr is that low-level design docs can be written as technotes, but that content should be deprecated and migrated into the software or system's user guide as it is implemented. There was a discussion about this on Community in relation to documenting FITS headers. The consensus seemed to be that since DM is an open source project, the code's documentation should be comprehensive enough to support users as both 'end users' and 'developers.'

jonathansick committed:
Copy edits

Fixes for https://github.com/lsst/LDM-493/pull/4#discussion_r85267679

jonathansick committed:
Fix: TCT, not CCB, approves LDM

Fixes https://github.com/lsst/LDM-493/pull/4#discussion_r85267571

jonathansick committed:
Remove claim that Git is better for collaboration

This section doesn't really *need* to make that claim.

https://github.com/lsst/LDM-493/pull/4#discussion_r85267769

TallJimbo commented on _platforms.rst:
```diff
@@ -0,0 +1,141 @@
+  This helps to make human-friendly URLs.
+  Audiences should see the ``lsst.io`` domain as a brand synonymous with LSST documentation.
+
+- `LTD publishes multiple versions of a document <https://sqr-006.lsst.io/#versioned-documentation-urls>`_, each under a different URL path prefix.
```
Do you mean `suffix`?

jonathansick replied:
I do mean prefix (for the path, not domain), as in

`/v/v1/`...

or `/v/DM-9999/` ...

jonathansick committed:
Don't link check ADS pages

http://adsabs.harvard.edu pages seem to be blocked, or non-responsive
from Travis.

jonathansick committed:
Add validation documents to the taxonomy

Validation documentation is still under-designed (the SDQA system is
currently undergoing a replan in LDM-151). Thus this mention of
validation documentation primarily serves as a placeholder. The details
that are mentioned, including that validations documents are generated
from automated systems, should be accurate, though.

Addresses https://github.com/lsst/LDM-493/pull/4#discussion_r85268282

jonathansick committed:
Typo fix

https://github.com/lsst/LDM-493/pull/5#issuecomment-258508955

jonathansick committed:
Merge branch 'tickets/DM-8099' into v1
